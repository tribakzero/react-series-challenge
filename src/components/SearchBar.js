import React from 'react';

const SearchBar = (props) => {
  return (
    <input onChange={props.onChange} />
  );
}

export default SearchBar;
