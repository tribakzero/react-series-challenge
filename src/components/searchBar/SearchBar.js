import React from 'react';
import { SearchBar as StyledSearchBar } from './SearchBar.style'

const SearchBar = (props) => {
  return (
    <StyledSearchBar>
      <input placeholder={(props.placeholder) ? props.placeholder : 'Buscar...'} onChange={props.onChange} />
    </StyledSearchBar>
  );
}

export default SearchBar;
