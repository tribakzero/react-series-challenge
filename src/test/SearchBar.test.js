import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from 'components/searchBar/SearchBar'

describe('SearchBar', () => {
  test('renders properly', () => {
    const component = shallow(<SearchBar />);
    expect(component).toMatchSnapshot()
  })

  test('receives onChange prop', () => {
    let wrapper = shallow(<SearchBar />)
    const mockChange = jest.fn();
    wrapper.setProps({ onChange: mockChange });
    wrapper.find('input').simulate(
      'change', { payload: 'test' }
    )
    expect(mockChange.mock.calls.length).toBe(1);
  });

  test('receives placeholder prop', () => {
    let wrapper = shallow(<SearchBar />)
    wrapper.setProps({ placeholder: 'TEST PLACEHOLDER' });
    expect(wrapper.find('input').prop('placeholder')).toBe('TEST PLACEHOLDER');
  });
});
