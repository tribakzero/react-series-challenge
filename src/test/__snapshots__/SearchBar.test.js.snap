// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <input
        onChange={undefined}
        placeholder="Buscar..."
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "onChange": undefined,
        "placeholder": "Buscar...",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <input
          onChange={undefined}
          placeholder="Buscar..."
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": undefined,
          "placeholder": "Buscar...",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
